crates/orderbook/src/orderbook.rs

```diff
@@ -483,62 +513,85 @@ mod tests {
         // App data does not encode cancellation.
         assert!(matches!(
             orderbook
-                .replace_order(
-                    old_order.metadata.uid,
-                    OrderCreation {
-                        from: Some(old_order.metadata.owner),
-                        signature: Signature::Eip712(Default::default()),
-                        ..Default::default()
-                    },
-                )
+                .replace_order(OrderCreation {
+                    from: Some(old_order.metadata.owner),
+                    signature: Signature::Eip712(Default::default()),
+                    ..Default::default()
+                },)
                 .await,
-            Err(ReplaceOrderError::InvalidReplacement)
+            Err(ReplaceOrderError::InvalidAppData(_))
         ));
 
         // Different owner
         assert!(matches!(
             orderbook
-                .replace_order(
-                    old_order.metadata.uid,
-                    OrderCreation {
-                        from: Some(H160([2; 20])),
-                        signature: Signature::Eip712(Default::default()),
-                        app_data: AppDataHash(cancellation.hash_struct()).into(),
-                        ..Default::default()
+                .replace_order(OrderCreation {
+                    from: Some(H160([2; 20])),
+                    signature: Signature::Eip712(Default::default()),
+                    app_data: OrderCreationAppData::Full {
+                        full: format!(
+                            r#"{{"version":"0.9.0","metadata":{{"replacedOrder":"{}"}}}}"#,
```

Comment:
Not really a must as the backend doesn't care about the version AFAICT, but just for the correctness sake, use version `1.1.0` where `repleaceOrder` is introduced.